File: RR.Application/Client/src/app/domain/models/rooms.ts
import { Manager } from "./manager";
import { Reservations } from "./reservations";
import { BaseModel } from './base-model';
import { Equipment, RoomEquipment, RoomDetails } from './equipment';

export interface Rooms extends BaseModel {
  Id: number;
  Name: string;
  Capacity: number;
  ManagerId: number;
  Manager: Manager;
  Reservations: Reservations[];
  RoomDetails: RoomDetails;
}

export type { RoomDetails, Equipment, RoomEquipment };

File: Core/Entities/Rooms.cs
using RR.Core.Entities.Base;

namespace RR.Core.Entities;

public class Rooms:BaseEntity 
{
    public int Id { get; set; }

    public string Name { get; set; }

    public int Capacity { get; set; }

    public int ManagerId { get; set; }

    //lazy loading
    public Manager Manager { get; set; }

    public RoomDetails RoomDetails { get; set; }

    public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
}
